packages/core/src/messagingProtocols/proteus/ProteusService/ProteusService.ts

```diff
@@ -173,4 +178,82 @@ export class ProteusService {
       state: response.errored ? PayloadBundleState.CANCELLED : PayloadBundleState.OUTGOING_SENT,
     };
   }
+
+  private async encryptPayloadForSession(
+    sessionId: string,
+    plainText: Uint8Array,
+  ): Promise<SessionPayloadBundle | undefined> {
+    this.logger.log(`Encrypting payload for session ID "${sessionId}"`);
+
+    let encryptedPayload: Uint8Array;
+
+    try {
+      const payloadAsArrayBuffer = await this.coreCryptoClient.proteusEncrypt(sessionId, plainText);
+      encryptedPayload = new Uint8Array(payloadAsArrayBuffer);
+    } catch (error) {
+      const notFoundErrorCode = 2;
+      if ((error as any).code === notFoundErrorCode) {
+        // If the session is not in the database, we just return undefined. Later on there will be a mismatch and the session will be created
+        return undefined;
+      }
+      this.logger.error(`Could not encrypt payload: ${(error as Error).message}`);
+      encryptedPayload = new Uint8Array(Buffer.from('💣', 'utf-8'));
+    }
+
+    return {encryptedPayload, sessionId};
+  }
+
+  public async encrypt(
+    plainText: Uint8Array,
+    users: UserPreKeyBundleMap | UserClients,
+    domain?: string,
+  ): Promise<{missing: UserClients; encrypted: OTRRecipients<Uint8Array>}> {
+    const encrypted: OTRRecipients<Uint8Array> = {};
+    const missing: UserClients = {};
+
+    for (const userId in users) {
+      const clientIds = isUserClients(users)
+        ? users[userId]
+        : Object.keys(users[userId])
+            // We filter out clients that have `null` prekey
+            .filter(clientId => !!users[userId][clientId]);
+      for (const clientId of clientIds) {
+        const sessionId = this.cryptographyService.constructSessionId(userId, clientId, domain);
+        const sessionExists = (await this.coreCryptoClient.proteusSessionExists(sessionId)) as unknown as boolean;
+        if (!sessionExists) {
+          const userQualifiedId = {id: userId, domain: domain ?? ''};
```

Comment:
What do we do when there's no domain?

Reply:
yep, empty string is the way to go 👌

Reply:
actually, you could just go `domain: string = ''` in the params of the function :)